Simplified/NYPLBookRegistry.m

```diff
@@ -347,7 +347,7 @@ - (void)syncWithCompletionHandler:(void (^)(BOOL success))completion
          }
          for (NSString *identifier in identifiersToRemove) {
            NYPLBookRegistryRecord *record = [self.identifiersToRecords objectForKey:identifier];
-           if (record.state & (NYPLBookStateDownloadSuccessful | NYPLBookStateUsed)) {
+           if (record.state == NYPLBookStateDownloadSuccessful || record.state == NYPLBookStateUsed) {
```

Comment:
there's an insidious *potential* bug here: if `record` happens to be nil, `record.state` will evaluate to 0 in objc. This means that depending on which rawValue the enum cases listed here evaluate to, this `if` condition could evaluate to true. This is not the case now, because the 0 case is mapped to Unregistered, but it would be more future-proof to add a nil check on record ANDed to what you have here.